io/src/csv/surface.hpp

```diff
@@ -0,0 +1,33 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// DFE include(s).
+#include <dfe/dfe_namedtuple.hpp>
+
+// System include(s).
+#include <cstdint>
+
+namespace traccc::io::csv {
+
+/// Type to read information into about the detector modules/surfaces
+struct surface {
+
+    uint64_t geometry_id = 0;
+    float cx = 0., cy = 0., cz = 0.;
+    float rot_xu = 0., rot_xv = 0., rot_xw = 0.;
+    float rot_yu = 0., rot_yv = 0., rot_yw = 0.;
+    float rot_zu = 0., rot_zv = 0., rot_zw = 0.;
```

Comment:
Do these defaults serve any purpose? Is it possible to read an object of this type without filling all its members with non-default values? :thinking:

Reply:
DFE is surprisingly robust with reading differently formatted files. If you read a file that doesn't provide all of these variables, DFE will silently just not touch the variables corresponding to the missing columns. I was actually looking for this type of a bug in my code for a few hours yesterday when by mistake a piece of code was reading an incorrect file. (Which resulted in some way more subtle errors than I would've expected.)

Long story short, setting some defaults helps at least a little with this...

Reply:
I'm not sure I would class "failing silently" as "surprisingly robust" but yeah that is a pretty good reason. :+1: